api/src/main/java/org/openmrs/module/appointmentscheduling/api/impl/AppointmentServiceImpl.java

```diff
@@ -1151,12 +1151,52 @@ public Appointment bookAppointment(Appointment appointment,
 		}
 
 		appointment.setStatus(AppointmentStatus.SCHEDULED);
-		return Context.getService(AppointmentService.class).saveAppointment(
+		Appointment savedAppnt = Context.getService(AppointmentService.class).saveAppointment(
 				appointment);
+		addNewStatusToHistory(savedAppnt);
+		return savedAppnt;
+	}
+
+	@Override
+	public AppointmentStatusHistory getPreviousAppointmentStatus(Appointment appointment) {
+		return appointmentStatusHistoryDAO.getPreviousAppointmentStatus(appointment);
+	}
+
+	@Override
+	@Transactional
+	public List<AppointmentStatusHistory> getAppointmentStatusHistories(Appointment appointment) {
+		return getAppointmentStatusHistoryDAO().getAppointmentStatusHistories(appointment);
+	}
+
+	@Override
+	public void addNewStatusToHistory(Appointment appointment) {
+		AppointmentStatusHistory previousStatus = getPreviousAppointmentStatus(appointment);
+		if (previousStatus != null) {
+			previousStatus.setEndDate(new Date());
```

Comment:
Shouldn't this check if the status has changed? Don't we want to create a new status if the current appointment status differs from the status returned by getPreviousAppoinmentStatus?